File: car-lease-service/src/main/java/org/jesperancinha/car/lease/model/Lease.java
package org.jesperancinha.car.lease.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.persistence.Table;

@Entity
@Table
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Lease {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @OneToOne
    private Car car;

    @OneToOne
    private Customer customer;

    @Column
    private Long duration;

    @Column
    private Long interestRate;

    @Column
    private Long netPrice;

   @Column
   private Double leaseRate;
}

File: car-lease-service/src/main/kotlin/org/jesperancinha/car/lease/model/Lease.kt
package org.jesperancinha.car.lease.model

import jakarta.persistence.*
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor

@Entity
@Table
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
class Lease {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private val id: Long? = null

    @OneToOne
    private val car: Car? = null

    @OneToOne
    private val customer: Customer? = null

    @Column
    private val duration: Long? = null

    @Column
    private val interestRate: Long? = null

    @Column
    private val netPrice: Long? = null

    @Column
    private val leaseRate: Double? = null
}